crates/router/src/services/conversion_impls.rs

```diff
@@ -0,0 +1,678 @@
+    format!("irrelevant {id_name} in {flow_name} flow")
+}
+fn get_default_router_data<F, Req, Resp>(
+    flow_name: &str,
+    request: Req,
+    response: Result<Resp, router_data::ErrorResponse>,
+) -> RouterData<F, Req, Resp> {
+    RouterData {
+        flow: std::marker::PhantomData,
+        merchant_id: get_irrelevant_id_string("merchant_id", flow_name),
+        customer_id: None,
+        connector_customer: None,
+        connector: get_irrelevant_id_string("connector", flow_name),
+        payment_id: get_irrelevant_id_string("payment_id", flow_name),
+        attempt_id: get_irrelevant_id_string("attempt_id", flow_name),
+        status: common_enums::AttemptStatus::default(),
+        payment_method: common_enums::PaymentMethod::default(),
+        connector_auth_type: router_data::ConnectorAuthType::default(),
+        description: None,
+        return_url: None,
+        address: PaymentAddress::default(),
+        auth_type: common_enums::AuthenticationType::default(),
+        connector_meta_data: None,
+        connector_wallets_details: None,
+        amount_captured: None,
+        access_token: None,
+        session_token: None,
+        reference_id: None,
+        payment_method_token: None,
+        recurring_mandate_payment_data: None,
+        preprocessing_id: None,
+        payment_method_balance: None,
+        connector_api_version: None,
+        request,
+        response,
+        connector_request_reference_id: get_irrelevant_id_string(
+            "connector_request_reference_id",
+            flow_name,
+        ),
+        #[cfg(feature = "payouts")]
+        payout_method_data: None,
+        #[cfg(feature = "payouts")]
+        quote_id: None,
+        test_mode: None,
+        connector_http_status_code: None,
+        external_latency: None,
+        apple_pay_flow: None,
+        frm_metadata: None,
+        dispute_id: None,
+        refund_id: None,
+        connector_response: None,
+        payment_method_status: None,
+        minor_amount_captured: None,
+    }
+}
+
+impl<T, Req: Clone, Resp: Clone> RouterDataConversion<T, Req, Resp> for AccessTokenFlowData {
+    fn from_old_router_data(
+        old_router_data: &RouterData<T, Req, Resp>,
+    ) -> errors::CustomResult<RouterDataV2<T, Self, Req, Resp>, errors::ConnectorError>
+    where
+        Self: Sized,
+    {
+        let resource_common_data = Self {};
+        Ok(RouterDataV2 {
+            flow: std::marker::PhantomData,
+            resource_common_data,
+            connector_auth_type: old_router_data.connector_auth_type.clone(),
+            request: old_router_data.request.clone(),
+            response: old_router_data.response.clone(),
+        })
+    }
+
+    fn to_old_router_data(
+        new_router_data: RouterDataV2<T, Self, Req, Resp>,
+    ) -> errors::CustomResult<RouterData<T, Req, Resp>, errors::ConnectorError>
+    where
+        Self: Sized,
+    {
+        let Self {} = new_router_data.resource_common_data;
```

Comment:
Is this line required, considering that there is nothing to destructure?

Reply:
Just in case someone adds a field to this struct in future.